fastlane_bot/events/exchanges/uniswap_v3.py

```diff
@@ -55,3 +55,16 @@ async def get_tkn0(self, address: str, contract: Contract, event: Any) -> str:
 
     async def get_tkn1(self, address: str, contract: Contract, event: Any) -> str:
         return await contract.caller.token1()
+
+    def get_pool_function(self, factory_contract: Contract):
+        """ Function to get pools from Factory.
+            This function is intended to be used with a Multicall. It fetches pools from a Uniswap V3 fork Factory contract.
+
+            Args:
+                factory_contract: The factory contract.
+
+            Returns:
+                The function.
+
+            """
+        return factory_contract.functions.getPool
```

Comment:
You may as well just use it directly where needed (since you're checking UniV3 before calling it).